### ISSUE TITLE
complementary-color-prediction demo error:  No data found for NDArray with id

### ISSUE THREAD

**Issue by QuantumInformation** [PAIR-code/deeplearnjs#500]
On my Mac,

When running `./scripts/watch-demo demos/complementary-color-prediction`

I get

![image](https://user-images.githubusercontent.com/216566/34472006-d8c4a9f0-ef4e-11e7-9dab-dc54448ee9a0.png)

**Comment by lunzhang** [comment]
I get the same error for `demos/nn-art`

**Comment by lunzhang** [comment]
I took a look at the cppn.ts source file and found that the NDArray
 ```
const inputAtlasWithLatentVariables =
        NDArray.make(this.inputAtlas.shape, {}) as Array2D;
```
was created with value of undefined therefore wasn't added to NDArrayMath.  

As a result `this.backend.getTexture(inputAtlasWithLatentVariables.id)` would throw the error.

**Comment by dsmilkov** [comment]
Thanks! We are on it. Fixed cppn - now looking at `demos/complementary-color-prediction` and other demos. Expect a PR soon (hopefully by end of day)

**Comment by dsmilkov** [comment]
This got fixed in #511
